packages/xo-web/src/common/tags.js

```diff
@@ -25,18 +46,82 @@ const ADD_TAG_STYLE = {
   verticalAlign: 'middle',
 }
 
-class SelectExistingTag extends Component {
-  state = { tags: [] }
+class AdvancedTagCreation extends Component {
+  state = {
+    tags: [],
+    tagConfigurations: this.props.tagConfigurations,
+  }
 
   get value() {
-    return this.state.tags.map(_ => _.value)
+    return { ...this.state, tags: this.state.tags.map(_ => _.value) }
+  }
+
+  onChangeTagConfiguration(tagId, key, value) {
+    const tagConfiguration = { ...this.state.tagConfigurations[tagId], ...{ [key]: value } }
+    const { tagConfigurations } = this.state
+    tagConfigurations[tagId] = tagConfiguration
+    this.setState({ tagConfigurations: { ...tagConfigurations } })
   }
 
   render() {
-    return <SelectTag multi onChange={this.linkState('tags')} value={this.state.tags} />
+    return (
+      <Container>
+        <Row className='d-flex'>
+          <Col>
+            <SelectTag multi onChange={this.linkState('tags')} value={this.state.tags} />
+          </Col>
+        </Row>
+        {this.props.isAdmin && (
+          <ul className='list-group'>
+            {this.state.tags.map(tag => {
+              const _onAddTagColor = () => this.onChangeTagConfiguration(tag.id, 'color', DEFAULT_TAG_COLOR)
+              const _onRemoveTagColor = () => this.onChangeTagConfiguration(tag.id, 'color', null)
+              const _onChangeTagColor = event => this.onChangeTagConfiguration(tag.id, 'color', getEventValue(event))
+
+              const tagConfiguration = this.state.tagConfigurations[tag.id]
+              return (
+                <li className='list-group-item' key={tag.id}>
+                  <Container>
+                    <Row className='d-flex'>
+                      <Col style={MARGIN_AUTO}>{tag.value}</Col>
+                      {tagConfiguration?.color == null ? (
+                        <Col style={TEXT_ALIGN_RIGHT}>
```

Comment:
Add `justify-content-between` to the `Row` should have the same effect as the 2 styles.

Reply:
`margin: auto` is required for the `tag.value` otherwise it is not vertically centered.

Reply:
Since the `add color` button is in a `Col`, the `space-between` will not be applied to the button but to the `Col`. The button will therefore not be on the `right`.